core/trino-main/src/main/java/io/trino/execution/MemoryRevokingScheduler.java

```diff
@@ -187,26 +189,29 @@ private void scheduleRevoking()
     private synchronized void runMemoryRevoking()
     {
         if (checkPending.getAndSet(false)) {
-            Collection<SqlTask> sqlTasks = null;
+            Collection<SqlTask> allTasks = null;
             for (MemoryPool memoryPool : memoryPools) {
                 if (!memoryRevokingNeeded(memoryPool)) {
                     continue;
                 }
 
-                if (sqlTasks == null) {
-                    sqlTasks = requireNonNull(currentTasksSupplier.get());
+                if (allTasks == null) {
+                    allTasks = requireNonNull(currentTasksSupplier.get());
                 }
 
-                requestMemoryRevoking(memoryPool, sqlTasks);
+                requestMemoryRevoking(memoryPool, allTasks);
```

Comment:
I thought it helped clarify the transformation state now that we don't re-traverse the tasks at once. `allTasks` has not been filtered until `findRunningTasksInMemoryPool` does the filter and sorting at which point you wouldn't want to accidentally traverse `allTasks` instead of `runningTasksInMemoryPool`. Granted, I also changed the type signatures to make that harder to do by accident but it did feel related to the refactor.